src/lamplib/src/genny/tasks/mothra_service.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+import subprocess
+import tempfile
+from typing import Optional
+import structlog
+import yaml
+
+MOTHRA_REPO = "git@github.com:10gen/mothra.git"
+TEAMS_DIR = "mothra/teams"
+TEAMS_FILES = ["cloud.yaml", "database.yaml", "rnd_dev_prod.yaml", "star.yaml"]
+TEAMS_FILES_PATHS = [f"{TEAMS_DIR}/{file}" for file in TEAMS_FILES]
+SLOG = structlog.get_logger(__name__)
+
+
+@dataclass
+class Team:
+    name: str
+    support_slack_channel_name: Optional[str] = None
+    support_slack_channel_id: Optional[str] = None
+
+    def __init__(self, **kwargs):
+        # Filter out any kwargs that are not in the class
+        filtered_kwargs = {k: v for k, v in kwargs.items() if k in self.__annotations__}
+
+        for k, v in filtered_kwargs.items():
+            setattr(self, k, v)
+
+
+class MothraService:
+
+    def __init__(self):
+        self.team_map = self._load_team_map()
+
+    def get_team(self, team_name: str) -> Optional[Team]:
+        return self.team_map.get(team_name, None)
+
+    def _load_team_map(self) -> dict[str, Team]:
+        SLOG.info("Cloning Mothra repository to get team information.")
+        with tempfile.TemporaryDirectory() as temp_dir:
+            subprocess.run(["git", "clone", MOTHRA_REPO, temp_dir])
```

Comment:
I like this approach, we may run into some perf issues and need to make this shallower or something, but this is easy enough to update later when we notice.

Reply:
Actually, upgrading this into a blocker. We can make this 3x quicker and save 100s of msecs.

```
$ time git clone git@github.com:10gen/mothra.git
Cloning into 'mothra'...
remote: Enumerating objects: 12393, done.
remote: Counting objects: 100% (1176/1176), done.
remote: Compressing objects: 100% (523/523), done.
remote: Total 12393 (delta 838), reused 947 (delta 647), pack-reused 11217
Receiving objects: 100% (12393/12393), 3.10 MiB | 2.89 MiB/s, done.
Resolving deltas: 100% (9220/9220), done.

________________________________________________________
Executed in    4.52 secs      fish           external
   usr time  384.12 millis    0.17 millis  383.96 millis
   sys time  197.60 millis    1.87 millis  195.73 millis
```

vs

```
$ time git clone --depth=1 git@github.com:10gen/mothra.git
Cloning into 'mothra'...
remote: Enumerating objects: 291, done.
remote: Counting objects: 100% (291/291), done.
remote: Compressing objects: 100% (277/277), done.
remote: Total 291 (delta 21), reused 140 (delta 7), pack-reused 0
Receiving objects: 100% (291/291), 634.19 KiB | 1.04 MiB/s, done.
Resolving deltas: 100% (21/21), done.

________________________________________________________
Executed in    3.93 secs      fish           external
   usr time  128.48 millis    0.17 millis  128.32 millis
   sys time  106.07 millis    1.71 millis  104.36 millis
```

These were repeatable results (M1 macbook, on a residential internet connection)